### DISCUSSION TITLE
RF 433 Mhz stopped working for Tasmota >9.1.0 (rc-switch library changed?)

### DISCUSSION THREAD

**Discussion by martin072** [arendst/Tasmota#13440] [Q&A] [answered]
I noticed when upgrading one of my Tasmota's from 9.1.0 to 9.4.0 (or 10.0.0), my 433Mhz devices stopped responding to commands send with Tasmota. 

Under 9.1.0 the following command works fine:
`cm?cmnd=RfSend 105588,18,8`

But with later versions this command does not trigger a response on the RF device anymore. Tried 9.4.0 and 10.0.0, both won't make the device to respond. Downgrading to 9.1.0 and all works fine. 

Any ideas?

**Comment by barbudor** [comment]
@martin072
There was some changes on Tasmota to support RF codes longer than 32 bits.
Could you try in between versions to narrow down the version where it broke?
Thanks

**Reply by martin072** [reply]
@barbudor 

9.2.0 worked fine as well, as of 9.3.0 it stops working for me.

**Reply by barbudor** [reply]
Can you try using `RfSend 105588,18,8,10,350` and `RfSend 105588,18,8,10,320` and report one or both are working ?

What are the devices ? 
In RC-Switch lib protocol 8 is listed as "08 (Came 12bit, HT12E)"

**Reply by martin072** [reply]
`RfSend 105588,18,8,10,350` is working, the other isn't. That is on version 9.3.0. 

Did I do something wrong?

**Comment by barbudor** [comment, marked as answer]
No you didn't made anything wrong
And your test helped me to clearly understand where is the problem
Up to 9.2.0 Tasmota was using a default pulse length of 350 when the parameter was not provided explicitly whatever was the protocol.
In Feb, following a issue reported by another user I made a change so that when the parameter is not provided we let the rc-switch library to decide which pulse length to use depending on the protocol used. For protocol 8, it is 320.
https://github.com/arendst/Tasmota/blob/dfafd34e55f00a1adaec0455ba72b01efbdc05a0/lib/lib_rf/rc-switch/src/RCSwitch.cpp#L123

For other protocols the default values varies between 20 and 700.

Going back to a fixed value of 350 is not possible as it would break other use cases
I see 2 possibilities:
1- changing rc-switch to have 350 as default for protocol 8 but I don't know what would be the impact for other devices that are using protocol 8.
2- no change in Tasmota and you change your work flow to the new command that is working

Would you be ok with option 2?

**Reply by martin072** [reply]
Thanks for the explanation. No change needed, I will stick with the new command, works fine for me!
